lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,139 @@
+from typing import Dict, List, Union
+from abc import ABC, abstractmethod
+
+import os
+from sycamore.data import Element, Document
+from sycamore.evaluation.pipeline import EvaluationPipeline
+from sycamore.evaluation import EvaluationDataPoint
+from sycamore.evaluation.metrics import document_retrieval_metrics, rouge_metrics
+from sycamore import Context
+
+
+def add_search_context_to_datapoint(datapoint: Document) -> List[Element]:
+    assert datapoint.type == "EvaluationDataPoint"
+    source_documents: List[Element] = []
+    assert datapoint.get("raw") is not None
+    for search_result in datapoint.get("raw", {}).get("SearchContexts", []):
+        source_document = Element()
+        pg_number = search_result.get("page_numbers", [search_result.get("page_number", None)])
+        assert pg_number is not None
+        properties = {
+            "_location": search_result.get("document_url", ""),
+            "page_number": pg_number[0],
+            "doc_id": search_result.get("document_id", [0]),
+        }
+        source_document.properties = properties
+        source_document.text_representation = search_result["text_representation"]
+        source_documents += [source_document]
+    return source_documents
+
+
+def add_filters_to_question(datapoint: EvaluationDataPoint) -> EvaluationDataPoint:
+    assert datapoint.type == "EvaluationDataPoint"
+    datapoint = EvaluationDataPoint(datapoint)
+    assert datapoint.raw is not None
+    template_Q = datapoint.raw["custom_question_augmentation"]
+    filter_Q = datapoint.raw["question_augmentation_filter"]
+    if datapoint.filters:
+        for filter in datapoint.filters.keys():
+            if datapoint.raw.get("Filters") is not None:
+                datapoint.filters[filter] = datapoint.raw.get("Filters").get(filter, "")
+        datapoint.ground_truth_source_documents = add_search_context_to_datapoint(datapoint)
+        datapoint.question = template_Q.format(datapoint.raw["Question"], datapoint.filters.get(filter_Q))
+    return datapoint
+
+
+class Assessment(ABC):
+    @abstractmethod
+    def run_evaluation(self, ctx: Context, index: str, **kwargs):
+        pass
+
+    def __call__(self, context: Context, index: str, **kwargs):
+        return self.run_evaluation(context, index, **kwargs)
+
+
+class QualityAssessment(Assessment):
+    def __init__(self, GT_path: str, rag_config: Dict, **kwargs):
```

Comment:
Please use standard Python field naming conventions, here it should be "gt_path" or "ground_truth_path".